src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx

```diff
@@ -1,56 +1,440 @@
+      })
+      const amountCryptoBaseUnit = toBaseUnit(amountCryptoPrecision, asset.precision)
+
+      // A THOR LP deposit can either be:
+      // - a RUNE MsgDeposit message type
+      // - an EVM custom Tx, i.e a Tx with calldata
+      // - a regular send with a memo (for ATOM and UTXOs)
+      const transactionType = (() => {
+        const supportedEvmChainIds = getSupportedEvmChainIds()
+        if (asset?.assetId === thorchainAssetId) return 'MsgDeposit'
+        if (supportedEvmChainIds.includes(fromAssetId(asset.assetId).chainId as KnownChainIds)) {
+          return 'EvmCustomTx'
+        }
+        return 'Send'
+      })()
+
+      await (async () => {
+        // We'll probably need to switch on chainNamespace instead here
+        switch (transactionType) {
+          case 'MsgDeposit': {
+            if (runeAccountNumber === undefined) throw new Error(`No account number found for RUNE`)
+
+            const adapter = assertGetThorchainChainAdapter()
+            const memo = `+:${thorchainNotationAssetId}:${otherAssetAddress ?? ''}:ss:${
+              confirmedQuote.feeBps
+            }`
+
+            const estimatedFees = await estimateFees({
+              cryptoAmount: amountCryptoBaseUnit,
+              assetId: asset.assetId,
+              memo,
+              to: THORCHAIN_POOL_MODULE_ADDRESS,
+              sendMax: false,
+              accountId,
+              contractAddress: undefined,
+            })
+
+            // LP deposit using THOR is a MsgDeposit tx
+            const { txToSign } = await adapter.buildDepositTransaction({
+              from: account,
+              accountNumber: runeAccountNumber,
+              value: amountCryptoBaseUnit,
+              memo,
+              chainSpecific: {
+                gas: (estimatedFees as FeeDataEstimate<KnownChainIds.ThorchainMainnet>).fast
+                  .chainSpecific.gasLimit,
+                fee: (estimatedFees as FeeDataEstimate<KnownChainIds.ThorchainMainnet>).fast.txFee,
+              },
+            })
+            const signedTx = await adapter.signTransaction({
+              txToSign,
+              wallet,
+            })
+            const txId = await adapter.broadcastTransaction({
+              senderAddress: account,
+              receiverAddress: THORCHAIN_POOL_MODULE_ADDRESS,
+              hex: signedTx,
+            })
+
+            setTxId(txId)
+
+            break
+          }
+          case 'EvmCustomTx': {
+            if (!inboundAddressData?.router) return
+            if (assetAccountNumber === undefined) return
+
+            const args = (() => {
+              const expiry = BigInt(dayjs().add(15, 'minute').unix())
+              const vault = getAddress(inboundAddressData.address)
+              const asset = isToken(fromAssetId(assetId).assetReference)
+                ? getAddress(fromAssetId(assetId).assetReference)
+                : '0x0000000000000000000000000000000000000000'
```

Comment:
TODO: sanity check me. If this is correct we should use a constant for the zero address.

Reply:
c9de4ee162 and added dev docs link comment while at it!

Reply:
Brilliant, thank you ser!